crates/rattler_installs_packages/src/resolve/dependency_provider.rs

```diff
@@ -57,12 +96,26 @@ impl VersionSet for PypiVersionSet {
     type V = PypiVersion;
 
     fn contains(&self, v: &Self::V) -> bool {
-        match (self.0.as_ref(), v) {
+        match (self.spec.as_ref(), v) {
             (Some(VersionOrUrl::Url(a)), PypiVersion::Url(b)) => a == b,
-            (Some(VersionOrUrl::VersionSpecifier(spec)), PypiVersion::Version(v)) => {
-                spec.contains(v)
+            (
+                Some(VersionOrUrl::VersionSpecifier(spec)),
+                PypiVersion::Version {
+                    version,
+                    only_prerelease,
+                },
+            ) => {
+                spec.contains(version)
+                    && (self.allows_prerelease || *only_prerelease || !version.any_prerelease())
```

Comment:
I have to try really hard to understand this boolean logic. Would you be able to comment it a little?